src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +215,21 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param types      The types of logs being requested.
+ * @param timeout    The timeout for getting the logs.  If the timeout expires, completion
+ *                   will be called with whatever logs have been retrieved by that point
+ *                   (which might be none, and might include partial logs).
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to pass in the file paths for the requested logs.
+ */
+- (void)getDiagnosticLogsOfTypes:(MTRDiagnosticLogTypes)types
+                         timeout:(NSTimeInterval * _Nullable)timeout
+                           queue:(dispatch_queue_t)queue
+                      completion:(void (^)(NSError * _Nullable error, NSArray<NSString *> * _Nullable filePathArray))completion MTR_PROVISIONALLY_AVAILABLE;
```

Comment:
So on timeout, will error be set?  But with a non-nil array?  I think we should have the second arg be a struct that has a "timed out" boolean and the array, and error not being set on timeout.  Or I guess go back to "you will get nothing on timeout"..

Reply:
IMO if someone is asking for logs with a timeout if i can't get you logs in that time, then there are no logs. With the partial logs or what we can that you suggested we might give them something that might or might not be valid. So I am not sure if that would be super useful.